### ISSUE TITLE
Merging option 'apollo' via Mixin breaks reactivity (apollo version 1.4.1)

### ISSUE THREAD

**Issue by tillkolter** [Akryum/vue-apollo#56]
I am sorry for the weird headline, I am not sure how to describe it better.

I have a mixin which defines a reusable apollo query:

```
export const ReleasePagingMixin = {
   apollo: {
    // Query with parameters
    releases: function () {
      return {
        query: gql`query Releases($first: Int!, $after: String) {
            releases(first: $first, after: $after) {
                ...Releases
            }
        }
        ${ReleaseList.fragments.releases}
        `,
}
...
```

Using the mixin works fine, but now I need to implement another query in the parent component:

```
<script>

  import {ReleasePagingMixin} from '../releases/releases-paging-mixin'
  import gql from 'graphql-tag'

  export default {
    components: {ReleaseList},
    mixins: [ReleasePagingMixin],
    apollo: {
      genre: function () {
        return {
          query: gql`query Genre($pk: ID!, $sub: Boolean) {
            genre(pk: $pk, sub: $sub) {
              name
            }
          }`,
          variables: {
            pk: <...ID...>
            sub: <...Boolean...>
          }
      }
   }
}
</script>
```

After the component has been rendered, both queries are executed, but nothing in the frontend is reactively rendered. 
I tried to debug a little and compare two version of my code (one which defines both methods directly under one option and the broken version which defines one query in the mixin and one in the parent component). It seems that in both versions, the queries are member of the parent component's `$apollo` instance, hence merged somehow correctly, but in the case of the seperate definitions, the `nextResult` function is not fired for any of the queries.

**Comment by Samuell1** [comment]
what is version of vue-apollo and apollo-client? I reported to @Akryum some mixins problem in newest version is fixed

**Comment by tillkolter** [comment]
I posted the version in the header, but not the dependencies. 
The problems I posted happend with the following versions:

apollo-client: 0.5.26
vue-apollo: 1.4.1

I think the problem is related to some conflict with `apollo-client/core/ObservableQuery`, but I dont come from a JS background and I am having problems analyzing the code yet. I was not sure to update to the current vue-apollo beta and take the risk to mess things up even more.
